local_planner/src/nodes/local_planner.cpp

```diff
@@ -167,6 +167,33 @@ void LocalPlanner::create2DObstacleRepresentation(const bool send_to_fcu) {
     updateObstacleDistanceMsg(to_fcu_histogram_);
   }
   polar_histogram_ = new_histogram;
+
+  //generate histogram image for logging
+  histogram_image_ = generateHistogramImage(polar_histogram_);
+}
+
+sensor_msgs::Image LocalPlanner::generateHistogramImage(Histogram& histogram) {
+
+	sensor_msgs::Image image;
+	double sensor_max_dist = 20.0;
+	image.header.stamp = ros::Time::now();
+	image.height = GRID_LENGTH_E;
+	image.width = GRID_LENGTH_Z;
+	image.encoding = sensor_msgs::image_encodings::MONO8;
+	image.is_bigendian = 0;
+	image.step = 255;
+
+	int image_size = (int)(image.height * image.width);
+	image.data.reserve(image_size);
+
+	//fill image data
+	for (int e = GRID_LENGTH_E - 1; e >= 0 ; e--) {
+	   for (int z = 0; z < GRID_LENGTH_Z; z++) {
```

Comment:
Any particular reason you forward-iterate for azimuth but reverse-iterate for elevation? I'm sure there's a reason, but it's not obvious so a comment about it might be nice

Reply:
Otherwise the image comes out upside down, since it is written in byte order, not addressed with x/y